### DISCUSSION TITLE
Use ESP_TX, ESP_RX in berry

### DISCUSSION THREAD

**Discussion by gonzzzales** [arendst/Tasmota#21525] [Q&A]
Hi!
I have NSPanel with Tasmota. Is it possible to use pins for flashing TX RT in my Berry-script? I want to connect one sensor through UART to NSPanel

**Comment by sfromis** [comment]
Assuming standard UART communication, you can simply use the `serial` module in Berry, and take it from there. You could do that with any available ESP32 gpio pin, does not have to be where the silk screen says something like tx/rx. Other types of serial usage can also be viable, like the serial bridge feature, allowing payloads to/from MQTT.

Some serial-connected sensors do already have support in Tasmota, without the need for coding anything.

**Comment by gonzzzales** [comment]
I just don't have any free GPIO ports, so i want to use port for flashing.
In Berry i initilize UART like this self.rs485 = serial(21, 19, 115200, serial.SERIAL_8N1) #21 - Rx, 19 - Tx

and what GPIO-ports have this UART?

**Reply by sfromis** [reply]
Probably tx/rx on gpio 1/3.

**Reply by gonzzzales** [reply]
thanks, i will try it
